model/publishing/test/TestBackupService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+declare(strict_types=1);
+
+namespace oat\taoPublishing\model\publishing\test;
+
+use Exception;
+use common_Exception;
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\import\service\RdsResourceNotFoundException;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoQtiTest\helpers\QtiPackageExporter;
+
+class TestBackupService extends ConfigurableService
+{
+    use OntologyAwareTrait;
+
+    public const FILESYSTEM_ID = 'publishing';
+
+    public const PROPERTY_QTI_TEST_BACKUP_PATH = 'http://www.tao.lu/Ontologies/TAOPublishing.rdf#QtiTestBackupPath';
+
+    /**
+     * @param string $deliveryUri
+     * @return File
+     * @throws RdsResourceNotFoundException
+     * @throws common_Exception
+     */
+    public function backupDeliveryTestPackage(string $deliveryUri): File
+    {
+        try {
+            $test = $this->getTestResource($deliveryUri);
```

Comment:
All you are using here is the test uri from the test resource, if it would be already in the event payload, you could spare some DB calls here. What do you think?

Reply:
Makes  sense. Created new PR to update event - https://github.com/oat-sa/extension-tao-delivery-rdf/pull/334